core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -32,7 +32,8 @@
 import org.apache.iceberg.relocated.com.google.common.collect.Iterables;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 
-class BaseSnapshot implements Snapshot {
+// visible for testing
+public class BaseSnapshot implements Snapshot {
```

Comment:
I don't love these changes, but they are in `core`.

Reply:
Do you mean that these are in `org.apache.iceberg` and the test is in `org.apache.iceberg.rest.responses`? All of the classes are in the core module.

I don't think there is a need for this change. Because we delegate to the `TableMetadataParser`, we just need to make sure that metadata makes it through correctly. We don't need to make a complicated `TableMetadata` instance. Instead, rely on `TableMetadataParser` to be correct. The metadata in should match the metadata coming out, but we don't want to do a ton of work to simply exercise `TableMetadataParser`.

Reply:
I did mean that they are all in `core`. As opposed to having this be in `api` where we have tests and where stability is guaranteed.

I agree with your statement though, and I will update the tests to read in a JSON file with a `TableMetadata` instance and then rely on `TableMetadataParser`, as we're testing `LoadTableResponse`.